File: Software Engineering/Java Web/MVC Frameworks - Spring/08. Spring Web API Introduction/Exercise Assignment/SpringStoreDemo/src/main/java/softuni/controllers/HomeController.java
package softuni.controllers;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;
import softuni.areas.games.models.view.HomePageGameView;
import softuni.areas.games.services.GameService;

import java.util.List;

@Controller
public class HomeController {
    private final GameService gameService;

    @Autowired
    public HomeController(GameService gameService) {
        this.gameService = gameService;
    }

    @GetMapping("/")
    public String index(Model model) {
        List<HomePageGameView> games = this.gameService.getAllHomePage();

        model.addAttribute("view", "home/index");
        model.addAttribute("title", "Home");
        model.addAttribute("games", games);

        return "base-layout";
    }
}

File: Software Engineering/C# Web/ASP.NET Core MVC June 2019/01. ASP.NET Core Introduction/Exercise/FDMC/Controllers/HomeController.cs
using Microsoft.AspNetCore.Mvc;
using FDMC.Services;

namespace FDMC.Controllers
{
    public class HomeController : Controller
    {
        private readonly IHomeService homeService;

        public HomeController(IHomeService homeService)
        {
            this.homeService = homeService;
        }

        public IActionResult Index()
        {
            return View(this.homeService.GetAllCats());
        }
    }
}
